fuzz/pkcs7_verify.cc

```diff
@@ -0,0 +1,96 @@
+GEludGVybmV0IFdpZGdpdHMgUHR5IEx0ZDAeFw0yMTA5MjExOTIyMTJaFw0yMjA5
+MjExOTIyMTJaMEUxCzAJBgNVBAYTAkFVMRMwEQYDVQQIDApTb21lLVN0YXRlMSEw
+HwYDVQQKDBhJbnRlcm5ldCBXaWRnaXRzIFB0eSBMdGQwggIiMA0GCSqGSIb3DQEB
+AQUAA4ICDwAwggIKAoICAQC1+MOn+BopcEVR4QMvjXdAxGkWFllXyQFDToL+qOiP
+RU1yN7C8KCtkbOAFttJIO4O/i0iZ7KqYbnmB6YUA/ONAcakocnrdoESgRJcVMeAx
+Dk/11OtMF5yIfeOOO/TUeVNmAUaT63gFbKy/adpqhzJtOv9BBl5VcYNGGSE+0wtb
+mjpmNsxunEQR1KLDc97fGYHeRfKoSyrCIEE8IaAEpKGR2Sku3v9Jwh7RpjupgiUA
+kH6pJk7VMZm5vl2wFjYvfysgjeN5ZtsxFDMaPYZStpxMxpNd5C9DsO2Ljp5NMpGf
+NGmG4ZqiaQg8z2cIM6ESmN1zDJdUh5IXed1fOxBZD/poUFH0wDRFWnvzlaPmjJEF
+rYLMK8svnE5nEQp9vu93ISFBx7cofs+niMaUXPEqaRSqruifN2M1it3kOf/8YZl1
+vurs+VtHD6nOJo6bd11+37aBidIB/BaWnzLrDmSTcPFa1tkTHwoLqc9+jThTq9jZ
+6w3lAMPpsoenyD19UmQB589+4kNp2SIO/TtzVQCGgQPXE2jDCl6G9aIPMkfvpPZK
+4THVil3WQRCFYnYdDO4HQXo2ZuC4RiqgY5ygfeoL+fa9k383lgxxAHQLS7xsbaVB
+40RmfdbdevgPYIwZNNO78ddRmMdSv6IknSW9gydGzY//btY+t1SWcBZWzn1Ewq8g
+2QIDAQABo1MwUTAdBgNVHQ4EFgQUotZD9ajEvnQYVezIWzcW4pzvMcUwHwYDVR0j
+BBgwFoAUotZD9ajEvnQYVezIWzcW4pzvMcUwDwYDVR0TAQH/BAUwAwEB/zANBgkq
+hkiG9w0BAQsFAAOCAgEAqCe42PIWoyLDx9bR+5cSp99N5xo5lLiSLtWx2emDbZB2
+AunqKYeEgIV+TWNF2w1SZ/ckFgV7SlL2Yl73N/veSNRfNAnpjLksGDFpdJb7YXrx
+cUvxdy1mr8oau6J7PC9JGjBTBrnhqwCQX1FtcAxODKll2Lsfuj6+bdC3rCK7KBEo
+ENamMJZIeo8lRP9qFF2xwCEzZjRv2zvB6O5o9045aTUcdCrwUfKE2sqY6EXRzFTC
+waK0HRCd1FLv9omhz/Ug5PMHP4d6MZfnAbFm+AzAhnpkrk/9TJYSOoNTNLWsuqhp
+dN0rKqiFWv1zIwfknXvTh1P1Ap+G5jffAca0zWUH1oKjE7ZZioSsaZ6gySnD8+WQ
+TPbOYtG+n0mhCH1TrU8Dqi3rd8g5IbC8loYLRH94QtodOnevD4Qo9Orfrsr8hGOW
+ABespanZArhoQ03DAtpNhtHm2NWJQF2uHNqcTrkq0omqZBTbMD1GKMBujoNooAUu
+w51U9r+RycPJTFqEGHb0nd7EjoyXEXtuX1Ld5fTZjQ9SszmQKQ8w3lHqRGNlkSiO
+e3IOOq2ruXmq1jykxpmi82IcTRUE8TZBfL/yz0nxpHKAYC1VwMezrkgZDGz4npxf
+1z2+qd58xU6/jsf7/+3xdPFubeEJujdbCkWQsQC5Rzm48zDWGq/pyzFji44K3TA=
+-----END CERTIFICATE-----
+)";
+
+OPENSSL_BEGIN_ALLOW_DEPRECATED
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t *buf, size_t len) {
+  BIO* data_bio = nullptr;
+  BIO *bio_cert = nullptr;
+  X509 *my_cert = nullptr;
+  X509_STORE *store = nullptr;
+  STACK_OF(X509) *certs = nullptr;
+  PKCS7 *pkcs7 = d2i_PKCS7(nullptr, &buf, len);
+  if (!pkcs7) {
+    goto end;
+  }
+  bio_cert = BIO_new_mem_buf(kCert, strlen(kCert));
+  if (!bio_cert) {
+    goto end;
+  }
+  my_cert = PEM_read_bio_X509(bio_cert, nullptr, nullptr, nullptr);
+  if (!my_cert) {
+    goto end;
+  }
+  store = X509_STORE_new();
+  if (!store) {
+    goto end;
+  }
+  if (!X509_STORE_add_cert(store, my_cert)) {
+    goto end;
+  }
+  certs = sk_X509_new_null();
+  if (!sk_X509_unshift(certs, my_cert)) {
+    goto end;
+  }
+
+  static const char kSignedData[] = "signed data";
+  data_bio = BIO_new_mem_buf(kSignedData, strlen(kSignedData));
+  if (!data_bio) {
+    goto end;
+  }
+
+  PKCS7_verify(pkcs7, certs, store, data_bio, nullptr, 0);
+
+end:
+  BIO_free(data_bio);
+  BIO_free(bio_cert);
+  X509_free(my_cert);
+  X509_STORE_free(store);
+  sk_X509_free(certs);
+  PKCS7_free(pkcs7);
+  return 0;
```

Comment:
Same comment regarding unique pointers from the other PR.

Reply:
Yeah. As mentioned in the other PR, the reason I'd prefer not to use smart pointers, and instead have explicit calls to the "xxx_free" functions, is to make it easier for us to debug/diagnose were a memory leak ever to surface. (Fuzz tests should always fail if there is a memory leak.)